src/main/java/io/gravitee/policy/assignmetrics/AssignMetricsPolicy.java

```diff
@@ -0,0 +1,174 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.policy.assignmetrics;
+
+import io.gravitee.gateway.api.ExecutionContext;
+import io.gravitee.gateway.api.Request;
+import io.gravitee.gateway.api.Response;
+import io.gravitee.gateway.api.buffer.Buffer;
+import io.gravitee.gateway.api.el.EvaluableRequest;
+import io.gravitee.gateway.api.el.EvaluableResponse;
+import io.gravitee.gateway.api.stream.BufferedReadWriteStream;
+import io.gravitee.gateway.api.stream.ReadWriteStream;
+import io.gravitee.gateway.api.stream.SimpleReadWriteStream;
+import io.gravitee.policy.api.PolicyChain;
+import io.gravitee.policy.api.annotations.OnRequest;
+import io.gravitee.policy.api.annotations.OnRequestContent;
+import io.gravitee.policy.api.annotations.OnResponse;
+import io.gravitee.policy.api.annotations.OnResponseContent;
+import io.gravitee.policy.assignmetrics.configuration.AssignMetricsPolicyConfiguration;
+import io.gravitee.policy.assignmetrics.configuration.PolicyScope;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Florent CHAMFROY (florent.chamfroy at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class AssignMetricsPolicy {
+
+    private static final Logger logger = LoggerFactory.getLogger(AssignMetricsPolicy.class);
+
+    private static final String REQUEST_VARIABLE = "request";
+    private static final String RESPONSE_VARIABLE = "response";
+
+    private final AssignMetricsPolicyConfiguration assignVariablePolicyConfiguration;
+
+    public AssignMetricsPolicy(final AssignMetricsPolicyConfiguration assignVariablePolicyConfiguration) {
+        this.assignVariablePolicyConfiguration = assignVariablePolicyConfiguration;
+    }
+
+    @OnRequestContent
+    public ReadWriteStream onRequestContent(Request request, ExecutionContext executionContext) {
+        if (
+            assignVariablePolicyConfiguration.getScope() != null &&
+            assignVariablePolicyConfiguration.getScope() == PolicyScope.REQUEST_CONTENT
```

Comment:
```suggestion
     PolicyScope.REQUEST_CONTENT.equals(assignVariablePolicyConfiguration.getScope())
```

OR (as == with enum is null safe)

```suggestion
    assignVariablePolicyConfiguration.getScope() == PolicyScope.REQUEST_CONTENT
```

Reply:
fix by 951af64